app/scripts/lib/pending-tx-tracker.js

```diff
@@ -101,13 +101,25 @@ module.exports = class PendingTransactionTracker extends EventEmitter {
     }))
   }
 
-  async _resubmitTx (txMeta) {
+  async _resubmitTx (txMeta, latestBlockNumber) {
+    if (!txMeta.firstRetryBlockNumber) {
+      this.emit('tx:block-update', txMeta, latestBlockNumber)
+    }
+
     if (Date.now() > txMeta.time + this.retryTimePeriod) {
       const hours = (this.retryTimePeriod / 3.6e+6).toFixed(1)
       const err = new Error(`Gave up submitting after ${hours} hours.`)
       return this.emit('tx:failed', txMeta.id, err)
     }
 
+    const firstRetryBlockNumber = txMeta.firstRetryBlockNumber
```

Comment:
this is undefined in QAing

Reply:
sorry one more thing... can you make this a conditional so `const firstRetryBlockNumber = txMeta.firstRetryBlockNumber || firstRetryBlockNumber`